src/ds/ds6/ds6-safety.cpp

```diff
@@ -184,4 +195,57 @@ namespace librealsense
 
         return results;
     }
+
+    void ds6_safety_sensor::set_safety_preset(int index, const rs2_safety_preset& sp) const
+    {
+        //// convert sp to vector of bytes
+        auto data_ptr = const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(&sp));
+
+        //check CRC
+        auto computed_crc32 = calc_crc32(data_ptr, sizeof(rs2_safety_preset));
+
+        // prepare vector of data
+        auto data_bytes = std::vector<uint8_t>(data_ptr, data_ptr + sizeof(sp));
+
+        // prepare header
+        rs2_safety_preset_header header = {1, 0xc0db, 405, computed_crc32 };
+        auto header_ptr = const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(&header));
```

Comment:
Not sure I understand :)
You cast to const and then `const_cast` it?
Maybe we can use `const rs2_safety_preset_header header` in the first place?

Reply:
changed the whole method, take a look on the new one when I commit the changes

Reply:
done.
optimized the whole function. take a look on the new one